cohortextractor/cohortextractor.py

```diff
@@ -774,20 +773,25 @@ def main():
         except DummyDataValidationError as e:
             print(f"Dummy data error: {e}")
             sys.exit(1)
-        except database_exceptions() as e:
+        except Exception as e:
             traceback.print_exc()
```

Comment:
I think it might be cleaner to just reraise non-database exceptions rather than try to handle them ourselves. So something like:
```py
if "DatabaseError" in str(e.__class__.mro()):
    traceback.print_exc()
    ...
    sys.exit(5)
else:
    raise
```

Or maybe even an early exit to avoid the indentation:
```py
# Ignore errors which don't look like database errors
if "DatabaseError" not in str(e.__class__.mro()):
    raise

traceback.print_exc()
sys.exit(5)
```

Reply:
That makes sense.  Does calling `raise` mean that it will exit with code 1 anyway?  I followed what was done for the `DummyDataValidation` block, but wondering if that should just call `raise` too?

Reply:
Yes, behaviourally it will be basically the same as what you already have: it will print a traceback and exit 1. So you definitely don't _have_ to make this change. 

The difference with the `DummyDataValidation` block is that it's only catching one class of exception, and we deliberately don't want to print the full trackback there — just the error itself — to avoid confusing users. So I think that can stay as it is.

The pattern I was suggesting here is effectively trying to do the same thing as the `DummyDataValidation` block. But, because we can't statically define the class of exceptions we want to catch, we have to catch everything and then re-raise if it turns out to be not the sort of exception we wanted to catch. (A bare `raise` statement is effectively saying "pretend I never caught this exception in the first place".)

Does that make sense?

Reply:
It does, thanks :slightly_smiling_face: I've made that change now too.